addons/mail/models/mail_message.py

```diff
@@ -1171,10 +1171,17 @@ def _notify_message_notification_update(self):
                     continue
                 else:
                     messages |= message
+        messages_per_partner = defaultdict(lambda: self.env['mail.message'])
+        for message in messages:
+            partner = self.env.user.partner_id
```

Comment:
I would name it current_user_partner, because partner is a too generic naming to be really helpful (or just don't use an intermediate variable, as you prefer)